File: test_integration.py
#!/usr/bin/env python3
"""
Test script to verify the FindPaper QA Engine integration
"""
import requests
import json

# Test configuration
BASE_URL = "http://localhost:8000"

def test_health():
    """Test health endpoint"""
    print("Testing health endpoint...")
    try:
        response = requests.get(f"{BASE_URL}/health")
        if response.status_code == 200:
            print("✅ Health check passed")
            print(f"Response: {response.json()}")
        else:
            print(f"❌ Health check failed: {response.status_code}")
    except Exception as e:
        print(f"❌ Health check error: {e}")

def test_qa_endpoint():
    """Test the integrated QA endpoint"""
    print("\nTesting QA endpoint...")
    
    test_query = {
        "query": "What are the latest advances in machine learning?",
        "limit": 10,
        "max_themes": 3,
        "model": None
    }
    
    try:
        response = requests.post(f"{BASE_URL}/qa", json=test_query)
        if response.status_code == 200:
            print("✅ QA endpoint test passed")
            result = response.json()
            print(f"Query: {result['original_query']}")
            print(f"Found {result['finding_info']['total_passages_found']} passages")
            print(f"Selected {result['qa_result']['processing_info']['selected_quotes']} quotes")
            print(f"Generated {result['qa_result']['processing_info']['themes_generated']} themes")
            print(f"Final report length: {len(result['qa_result']['final_report'])} characters")
        else:
            print(f"❌ QA endpoint test failed: {response.status_code}")
            print(f"Error: {response.text}")
    except Exception as e:
        print(f"❌ QA endpoint error: {e}")

def test_scholarqa_endpoints():
    """Test individual ScholarQA endpoints"""
    print("\nTesting ScholarQA endpoints...")
    
    # Test QA pipeline endpoint
    test_data = {
        "query": "What is artificial intelligence?",
        "ranked_passages": [
            {
                "paper_id": "test_1",
                "title": "Test Paper 1",
                "evidence": "Artificial intelligence is the simulation of human intelligence in machines.",
                "cross_score": 0.8,
                "final_score": 0.9
            },
            {
                "paper_id": "test_2", 
                "title": "Test Paper 2",
                "evidence": "Machine learning is a subset of artificial intelligence.",
                "cross_score": 0.7,
                "final_score": 0.8
            }
        ],
        "max_themes": 2
    }
    
    try:
        response = requests.post(f"{BASE_URL}/qa-pipeline", json=test_data)
        if response.status_code == 200:
            print("✅ ScholarQA pipeline endpoint test passed")
            result = response.json()
            print(f"Generated {len(result['themes'])} themes")
            print(f"Final report length: {len(result['final_report'])} characters")
        else:
            print(f"❌ ScholarQA pipeline endpoint test failed: {response.status_code}")
            print(f"Error: {response.text}")
    except Exception as e:
        print(f"❌ ScholarQA pipeline endpoint error: {e}")

if __name__ == "__main__":
    print("🚀 Starting FindPaper QA Engine Integration Tests")
    print("=" * 50)
    
    test_health()
    test_qa_endpoint()
    test_scholarqa_endpoints()
    
    print("\n" + "=" * 50)
    print("🏁 Integration tests completed")
    print("\nNote: Make sure the server is running with: python app.py")

File: frontend/test-integration.js
#!/usr/bin/env node

/**
 * Simple integration test for FindPaper QA Engine Frontend
 * Tests the API connection and basic functionality
 */

const API_BASE_URL = process.env.NEXT_PUBLIC_API_URL || 'http://127.0.0.1:8000'

async function testHealthCheck() {
  console.log('🔍 Testing health check...')
  try {
    const response = await fetch(`${API_BASE_URL}/health`)
    if (response.ok) {
      const data = await response.json()
      console.log('✅ Health check passed:', data)
      return true
    } else {
      console.log('❌ Health check failed:', response.status)
      return false
    }
  } catch (error) {
    console.log('❌ Health check error:', error.message)
    return false
  }
}

async function testQAEndpoint() {
  console.log('\n🔍 Testing QA endpoint...')
  try {
    const testQuery = {
      query: "What are the latest advances in machine learning?",
      limit: 10,
      max_themes: 3,
      model: "gpt-4o-mini"
    }

    const response = await fetch(`${API_BASE_URL}/qa`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(testQuery)
    })

    if (response.ok) {
      const data = await response.json()
      console.log('✅ QA endpoint test passed')
      console.log(`📊 Found ${data.finding_info.total_passages_found} passages`)
      console.log(`📝 Selected ${data.qa_result.processing_info.selected_quotes} quotes`)
      console.log(`🎯 Generated ${data.qa_result.processing_info.themes_generated} themes`)
      console.log(`📄 Answer length: ${data.qa_result.final_report.length} characters`)
      return true
    } else {
      const errorText = await response.text()
      console.log('❌ QA endpoint test failed:', response.status, errorText)
      return false
    }
  } catch (error) {
    console.log('❌ QA endpoint error:', error.message)
    return false
  }
}

async function testSearchPassages() {
  console.log('\n🔍 Testing search passages endpoint...')
  try {
    const testQuery = {
      query: "machine learning",
      limit: 5
    }

    const response = await fetch(`${API_BASE_URL}/search_passsages`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(testQuery)
    })

    if (response.ok) {
      const data = await response.json()
      console.log('✅ Search passages test passed')
      console.log(`📊 Found ${data.matched_count} passages`)
      return true
    } else {
      const errorText = await response.text()
      console.log('❌ Search passages test failed:', response.status, errorText)
      return false
    }
  } catch (error) {
    console.log('❌ Search passages error:', error.message)
    return false
  }
}

async function runTests() {
  console.log('🚀 Starting FindPaper QA Engine Frontend Integration Tests')
  console.log('=' .repeat(60))
  console.log(`🌐 API Base URL: ${API_BASE_URL}`)
  console.log('=' .repeat(60))

  const results = {
    health: await testHealthCheck(),
    qa: await testQAEndpoint(),
    search: await testSearchPassages()
  }

  console.log('\n' + '=' .repeat(60))
  console.log('📊 Test Results Summary:')
  console.log('=' .repeat(60))
  console.log(`Health Check: ${results.health ? '✅ PASS' : '❌ FAIL'}`)
  console.log(`QA Endpoint: ${results.qa ? '✅ PASS' : '❌ FAIL'}`)
  console.log(`Search Passages: ${results.search ? '✅ PASS' : '❌ FAIL'}`)
  
  const allPassed = Object.values(results).every(result => result)
  console.log('\n' + '=' .repeat(60))
  console.log(`Overall Result: ${allPassed ? '🎉 ALL TESTS PASSED' : '⚠️  SOME TESTS FAILED'}`)
  console.log('=' .repeat(60))

  if (!allPassed) {
    console.log('\n💡 Troubleshooting Tips:')
    console.log('1. Make sure the backend is running: python app.py')
    console.log('2. Check if the API URL is correct:', API_BASE_URL)
    console.log('3. Verify environment variables are set correctly')
    console.log('4. Check backend logs for any errors')
  }

  process.exit(allPassed ? 0 : 1)
}

// Run tests if this script is executed directly
if (require.main === module) {
  runTests().catch(console.error)
}

module.exports = { testHealthCheck, testQAEndpoint, testSearchPassages }
